### ISSUE TITLE
"openssl x509 -in <pem cert-file> -checkend <seconds>" does not work

### ISSUE THREAD

**Issue by RoMo17** [openssl/openssl#4769]
No matter how grate number of seconds we check, the result is the message "Certificate will not expire" shows, and "echo $?" shows a zero (0).

We have tried on AIX 7.2 and some Linux version.

Is there a limit for the number of seconds that can be used? If so, why do we not gett an error message about that? 

> openssl x509 -noout  -in /read/ssl/read-cert.pem  -enddate
notAfter=Apr  7 21:58:01 2019 GMT

> date
Tue Nov 21 16:18:42 CET 2017

~ 502 days to end of validity.

To see if the cert will become invalid within 505 days:

> openssl x509  -in /read/ssl/read-cert.pem  -checkend $( expr 24 \\* 60 \\* 60 \\* 60 \\* 505 ) ; echo $?
Certificate will not expire
0

**Comment by RoMo17** [comment]
The openssl versions wopenssl versions we  have tested:

> OpenSSL 1.0.2h  3 May 2016
 
on AIX and Linux

and
> OpenSSL 1.0.2k  26 Jan 2017 

in Cygwin on Windows 7

**Comment by t-j-h** [comment]
Try the following:

openssl x509 -in /read/ssl/read-cert.pem -checkend $( expr 24 \* 60 \* 60 \* 505 ) ; echo $?

**Comment by RoMo17** [comment]
Yes, I understand that I was very generous with the 'seconds' ;-)
But that only made it even more secure that the certificate would become invalid within that period.

**Comment by t-j-h** [comment]
No - you will have overflowed the time range supported on the platform if your time value is stored in a 32-bit signed quantity.

**Comment by RoMo17** [comment]
Thank you!
So, that means that the numer of secs must be less than 2,147,483,647  ,  24,855 days ?

Hm... should be sufficient ;-)

**Comment by RoMo17** [comment]
Would be nice if there was an error handling that gave an error message when there is an overflow, of any kind.

**Comment by ghadf** [comment]
56087077d81e2b888f4cbe7f70b2077dc5add90d seems to add a check that the argument does not overflow time_t, but (if I'm not mistaken) undetected overflow may also occur when computing tcheck.

**Comment by sreedhar-b7** [comment]
I cannot understand what I encircled in the attachment in the patch . Why checkoffset is compared with checkoffset itself when the type of this variable is also time_t?
<img width="576" alt="patch_checkend" src="https://github.com/openssl/openssl/assets/76534813/f667725a-ddff-49f2-92e6-a82c657148b0">
